internal/app.go

```diff
@@ -319,29 +319,32 @@ func (a *App) runOnChanged(fname string, fullpath string, onChanged []string) {
 
 func (a *App) checkin(client *http.Client, crypto CryptoHandler) error {
 	headers := make(map[string]string)
+	var config configSnapshot
+	var err error
 
 	if fi, err := os.Stat(a.EncryptedConfig); err == nil {
 		// Don't pull it down unless we need to
 		ts := fi.ModTime().UTC().Format(time.RFC1123)
 		headers["If-Modified-Since"] = ts
 	}
 
+	if config.prev, err = UnmarshallFile(nil, a.EncryptedConfig, false); err != nil {
+		var perr *os.PathError
+		if !errors.As(err, &perr) || !os.IsNotExist(perr) {
+			log.Printf("Unable to load previous config version: %s. Forcing config update", err)
+			delete(headers, "If-Modified-Since")
+		}
+	}
```

Comment:
Can we combine these two if statements like below?
```
	if config.prev, err = UnmarshallFile(nil, a.EncryptedConfig, false); err != nil {
		var perr *os.PathError
		if !errors.As(err, &perr) || !os.IsNotExist(perr) {
			log.Printf("Unable to load previous config version: %s. Forcing config update", err)
		}
	}
	if config.prev != nil {
		if fi, err := os.Stat(a.EncryptedConfig); err == nil {
			// Don't pull it down unless we need to
			ts := fi.ModTime().UTC().Format(time.RFC1123)
			headers["If-Modified-Since"] = ts
		}
	}
```

That just looks cleaner to me.